src/pages/saksbehandling/steg/uførhet/Uførhet.tsx

```diff
@@ -96,38 +70,48 @@ const Uførhet = (props: VilkårsvurderingBaseProps) => {
     const handleSave = async (values: FormData, nesteUrl: string) => {
         if (!values.status) return;
 
-        const uføreValues: UførhetType = {
-            status: values.status,
-            uføregrad: values.uføregrad ? parseInt(values.uføregrad, 10) : null,
-            forventetInntekt: values.forventetInntekt ? parseInt(values.forventetInntekt, 10) : null,
-            begrunnelse: values.begrunnelse,
-        };
+        const isEqual = (): boolean =>
+            /* Bruker == istedenfor === siden `undefined == null => true` */
+            values.status == props.behandling.vilkårsvurderinger.uføre?.vurdering?.resultat &&
+            values.uføregrad == props.behandling.vilkårsvurderinger.uføre?.vurdering?.grunnlag?.uføregrad &&
+            values.forventetInntekt ==
+                props.behandling.vilkårsvurderinger.uføre?.vurdering?.grunnlag?.forventetInntekt &&
+            values.begrunnelse == props.behandling.vilkårsvurderinger.uføre?.vurdering.begrunnelse;
```

Comment:
Jeg føler det blir skummelt med èn gang vi må ty til `==`. Det er også litt vanskelig å henge med på at det vi _egentlig_ sjekker er om noen av verdiene har endret seg fra defaulten(?).

Hadde det gått an å dra ut en funksjon som tok inn f.eks. vurderinga og returnerte default-verdiene til formet? Og så for denne sjekket likhet på nåværende formverdier og det som var default-verdiene?

Reply:
Smartere det ja